### PR TITLE
improve type checking and update version

### PR DESCRIPTION
**Pull request by mustafa-cosar**

### CODE REVIEW

File: src/stringify.js [hunk 1]
```diff
@@ -75,6 +76,29 @@ var JSON = module.exports;
...
+};
+const INT64_KEYS = ["buffer", "offset"];
+
+const ARRAY_REPRESENTATION = "[object Array]";
+const NULL = "null";
+
+const UNICODE_PADDING = "0000";
+const UNICODE_LENGTH = 4;
+
+const SUBSTITUTION_TABLE = {
```
Comment (thehcma) [inline src/stringify.js]: CHARACTER_2_ESCAPED_CHARACTER

File: src/stringify.js [hunk 2]
```diff
@@ -104,10 +119,10 @@ var BASE_16 = 16;
         escapable.lastIndex = 0;
         return escapable.test(string) ? '"' + string.replace(escapable, function (a)
         {
-            var c = meta[a];
-            return typeof c === 'string' ?
+            var c = SUBSTITUTION_TABLE[a];
+            return typeof c === constnts.STRING_TYPE ?
```
Comment (thehcma) [inline src/stringify.js]: constants

Comment (thehcma) [reply src/stringify.js]: btw, didn't this break anything?

File: test/jsonInt64.spec.js [hunk 3]
```diff
@@ -66,4 +66,22 @@ describe("Tests the Int64 support", function ()
...
+        var byteArray = new Uint8Array([0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff]);
+        var offset = 0;
+
+        var int64Like = {buffer: byteArray, offset: offset};
+
+        expect(int64Like instanceof Int64).to.be.false;
+        expect(JSONInt64.stringify(int64Like)).to.be.equal(maxSignedInt64DecimalString);
+        expect(JSONInt64.parse(JSONInt64.stringify(int64Like))).to.be.equalInt64(maxSignedInt64DecimalInt64);
+
+        var nonInt64Like1 = {buffer: byteArray, offset: "0"};
```
Comment (thehcma) [inline test/jsonInt64.spec.js]: why the 1 suffix?

**Review by thehcma — CHANGES_REQUESTED** [review]
Please look into the "constants" typo - should it not have broken the build tests?

File: src/stringify.js (hunk 1 above)
Comment (mustafa-cosar) [reply src/stringify.js]: fixed

File: test/jsonInt64.spec.js (hunk 3 above)
Comment (mustafa-cosar) [reply test/jsonInt64.spec.js]: fixed

File: src/stringify.js (hunk 2 above)
Comment (mustafa-cosar) [reply src/stringify.js]: you're right. Let me write a failing test.

**Comment by mustafa-cosar** [comment]
I added a lot of tests

File: test/jsonInt64.spec.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
-var expect = require('chai').expect;
-var Assertion = require('chai').Assertion;
+var chai = require('chai');
+var Assertion = chai.Assertion;
```
Comment (thehcma) [inline test/jsonInt64.spec.js]: I am not seeing ``Assertion`` used anywhere? (at least not in this changeset - is it needed?)

Comment (thehcma) [inline test/jsonInt64.spec.js:2]: I am not seeing ``Assertion`` used anywhere in this changeset - is it really needed?

File: test/jsonInt64.spec.js [hunk 5]
```diff
@@ -80,8 +82,115 @@ describe("Tests the Int64 support", function ()
...
+            scientific: 1.2e+23
+        }
+
+        expect(JSONInt64.stringify(complexObject)).to.be.equal(complexObjectAsString);
+        expect(JSONInt64.parse(complexObjectAsString)).to.be.deep.equal(complexObjectAfterParse);
+
+        done();
+    });
+
+    it("tests that fromDecimalString throws error for too long and too big strings", function (done)
```
Comment (thehcma) [inline test/jsonInt64.spec.js:148]: error -> an error

**Review by thehcma — APPROVED** [review]

File: test/jsonInt64.spec.js (hunk 5 above)
Comment (mustafa-cosar) [reply test/jsonInt64.spec.js:148]: fixed

File: test/jsonInt64.spec.js (hunk 4 above)
Comment (mustafa-cosar) [reply test/jsonInt64.spec.js:2]: I use it to add a custom equality checker for `Int64`

**Pull request merged**